File: java/leetcode_grind/Day637.java
package leetcode_grind;

public class Day637 {
    // https://leetcode.com/problems/lemonade-change/description/
    static class Solution1 {
        public boolean lemonadeChange(int[] bills) {
            var fiveBills = 0;
            var tenBills = 0;
            for (var b : bills) {
                if (b == 5) {
                    fiveBills++;
                } else if (b == 10) {
                    if (fiveBills == 0) {
                        return false;
                    }
                    fiveBills--;
                    tenBills++;
                } else {
                    if (fiveBills > 0 && tenBills > 0) {
                        fiveBills--;
                        tenBills--;
                    } else if (fiveBills >= 3) {
                        fiveBills -= 3;
                    } else {
                        return false;
                    }
                }
            }
            return true;
        }
    }
}

File: src/leetcode_grind/day_637.rs
// https://leetcode.com/problems/lemonade-change/description/
pub fn lemonade_change(bills: Vec<i32>) -> bool {
    let mut five_dollars_bills = 0;
    let mut ten_dollars_bills = 0;

    for customer_bill in bills {
        if customer_bill == 5 {
            five_dollars_bills += 1;
        } else if customer_bill == 10 {
            if five_dollars_bills > 0 {
                five_dollars_bills -= 1;
                ten_dollars_bills += 1;
            } else {
                return false;
            }
        } else {
            if ten_dollars_bills > 0 && five_dollars_bills > 0 {
                five_dollars_bills -= 1;
                ten_dollars_bills -= 1;
            } else if five_dollars_bills >= 3 {
                five_dollars_bills -= 3;
            } else {
                return false;
            }
        }
    }
    true
}
